sktime/transformations/base.py

```diff
@@ -135,6 +135,8 @@ class BaseTransformer(BaseEstimator):
         # todo: rename to capability:missing_values
         "capability:missing_values:removes": False,
         # is transform result always guaranteed to contain no missing values?
+        "capability:categorical_in_X": False,
```

Comment:
do we not want the default to be `True`? That would be more flexible to users and developers.

Reply:
I didn't get you.

Default is False since most estimators cannot handle natively. Rather we switch it on by setting tag to True in the few that can support it. This is how we did it in forecasting as well.

Reply:
My thinking behind the comment, the `True` setting "lets more things through". This is a trade-off in blocking estimators where things would already work out of the box - e.g., in the many cases where all we do is set the tag to `True` - vs having uninformative errors.

Do we really want to go through all the composites one-by-one, there are so many.

Reply:
(my opinion on this is not too strong, I just think it's worth a discussion)

Reply:
Setting default to `True` will lead to uninformative errors in all the estimators that do not support. I think it is safer to keep the default as `False` and add capability separately by setting it to `True` in supporting estimators.

Reply:
this is a "safety" vs "convenience" tradeoff in my opinion. so that's not the only argument to consider... as said, no strong opinion.